src/api/search.js

```diff
@@ -1,31 +1,39 @@
 import { cachedFetch, buildMwApiUrl } from 'utils'
 
 export const search = (lang, term) => {
+  // fulltext search
   const params = {
     action: 'query',
-    prop: ['description', 'pageimages', 'pageprops'].join('|'),
+    prop: 'pageimages',
+    generator: 'search',
+    list: 'search',
+    srprop: 'snippet',
+    srsearch: term,
+    srlimit: 15,
+    srenablerewrites: true,
+    srinfo: 'rewrittenquery',
+    gsrsearch: term,
+    gsrnamespace: 0,
+    gsrlimit: 15,
     piprop: 'thumbnail',
-    pilimit: 15,
-    ppprop: 'displaytitle',
-    generator: 'prefixsearch',
-    redirects: true,
     pithumbsize: 64,
-    gpslimit: 15,
-    gpsnamespace: 0,
-    gpssearch: term.replace(/:/g, ' ')
+    pilimit: 15,
+    format: 'json'
   }
+
   const url = buildMwApiUrl(lang, params)
   return cachedFetch(url, data => {
-    if (!data.query || !data.query.pages) {
+    if (!data.query || !data.query.search) {
       return []
     }
-    data.query.pages.sort((a, b) => a.index - b.index)
```

Comment:
@stephanebisson do you still remember why sorting by index here?

Reply:
It looks like the results have an order based on relevance but they were not ordered inside the json response.

Reply:
Thanks, fulltext search has it (relevance) by default, so safe to remove.